src/uu/uniq/src/uniq.rs

```diff
@@ -207,76 +220,345 @@ impl Uniq {
         }
 
         if self.show_counts {
-            write!(writer, "{count:7} {line}")
+            let prefix = format!("{count:7} ");
+            let out = prefix
+                .as_bytes()
+                .iter()
+                .chain(line.iter())
+                .copied()
+                .collect::<Vec<u8>>();
+            writer.write_all(out.as_slice())
         } else {
-            writer.write_all(line.as_bytes())
+            writer.write_all(line)
         }
         .map_err_context(|| "Failed to write line".to_string())?;
 
         write_line_terminator!(writer, line_terminator)
     }
 }
 
-fn get_line_string(io_line: io::Result<Vec<u8>>) -> UResult<String> {
-    let line_bytes = io_line.map_err_context(|| "failed to split lines".to_string())?;
-    String::from_utf8(line_bytes)
-        .map_err(|e| USimpleError::new(1, format!("failed to convert line to utf8: {e}")))
+fn opt_parsed(opt_name: &str, matches: &ArgMatches) -> UResult<Option<usize>> {
```

Comment:
I think it can be simplify this way:

fn opt_parsed(opt_name: &str, matches: &ArgMatches) -> UResult<Option<usize>> {
    matches.get_one::<String>(opt_name)
        .map(|arg_str| {
            arg_str.parse::<usize>()
                .map_err(|e| match e.kind() {
                    IntErrorKind::PosOverflow => USimpleError::new(1, format!("{} is too large for {}", arg_str.maybe_quote(), opt_name)),
                    _ => USimpleError::new(1, format!("Invalid argument for {}: {}", opt_name, arg_str.maybe_quote())),
                })
                .map(Some)
        })
        .transpose()
}

Reply:
well, not exactly - we need to convert one of the error cases (IntErrorKind::PosOverflow) from parse() into a value of usize::MAX , while passing all other errors on

Reply:
ok, sorry